manifest: Update west manifest to add LVGL shell support

faxe1008 opened the pull request:
Updates the west manifest to add an LVGL shell with initial memory statistics and monkey testing support.

Relates to [lvgl/pull/42](https://github.com/zephyrproject-rtos/lvgl/pull/42)

faxe1008 committed:
manifest: Update west manifest to add LVGL shell support

Updates the west manifest to add an LVGL shell with initial memory
statistics and monkey testing support.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

fabiobaltieri commented:
@faxe1008 should this PR also have a `CONFIG_LV_Z_SHELL=y` in the LVGL sample so the new code gets built in CI?

faxe1008 commented:
@fabiobaltieri Good catch :^).  would you have it enabled by default or added only  for e.g. in `native_posix` board conf?

fabiobaltieri commented:
> @fabiobaltieri Good catch :^). would you have it enabled by default or added only for e.g. in `native_posix` board conf?

I think it would make sense to enable it in `samples/subsys/display/lvgl/prj.conf`, so all platform when building that sample.

faxe1008 committed:
samples: lvgl: Enable lvgl shell

Activate the lvgl shell with memory statistics and monkey testing support
for the lvgl sample.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

jfischer-no commented on samples/subsys/display/lvgl/prj.conf:
```diff
@@ -12,4 +13,5 @@ CONFIG_LV_USE_LOG=y
 CONFIG_LV_USE_LABEL=y
 CONFIG_LV_USE_BTN=y
 CONFIG_LV_USE_IMG=y
+CONFIG_LV_USE_MONKEY=y
```
How does it affect flash memory usage?

faxe1008 replied:
The `monkey` is an opaque structure and is allocated within the memory pool the zephyr manages for lvgl.
So the only overhead is actual code size for that feature (and the shell commands).

Built the sample for `mimxrt1060_evk`.
Without `CONFIG_LV_USE_MONKEY`:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      255856 B         8 MB      3.05%
             RAM:     1077224 B        32 MB      3.21%
            ITCM:          0 GB       128 KB      0.00%
            DTCM:          0 GB       128 KB      0.00%
           OCRAM:          0 GB       768 KB      0.00%
        IDT_LIST:          0 GB         2 KB      0.00%
```

With `CONFIG_LV_USE_MONKEY=y`:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      257720 B         8 MB      3.07%
             RAM:     1077224 B        32 MB      3.21%
            ITCM:          0 GB       128 KB      0.00%
            DTCM:          0 GB       128 KB      0.00%
           OCRAM:          0 GB       768 KB      0.00%
        IDT_LIST:          0 GB         2 KB      0.00%
```
`257720-255856=1864`, so roughly `1KiB`.

fabiobaltieri commented on west.yml:
```diff
@@ -260,7 +260,7 @@ manifest:
       path: modules/lib/loramac-node
     - name: lvgl
-      revision: f7cf01d413aa9e76f376e25aa7b5c08dfdbc3c24
+      revision: pull/42/head
```
5da257f782a8f9c6e265bdc60ebc2a93fdee24de all yours

faxe1008 replied:
Update the revision :^)
